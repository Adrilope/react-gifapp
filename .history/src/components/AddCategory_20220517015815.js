import React from 'react'

export const AddCategory = () => {
  return (
    <>
    <h2></h2>AddCategory</>
  )
}
