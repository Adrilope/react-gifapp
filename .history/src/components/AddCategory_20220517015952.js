import React from 'react'

export const AddCategory = () => {
    return (
        <>
            <h3>Add Category</h2>
        </>
    )
}
