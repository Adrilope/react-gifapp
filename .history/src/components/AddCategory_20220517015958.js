import React from 'react'

export const AddCategory = () => {
    return (
        <>
            <h3>Add Category</h3>
            <input
        </>
    )
}
