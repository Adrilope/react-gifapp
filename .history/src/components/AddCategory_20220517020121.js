import React from 'react'

export const AddCategory = () => {

    co

    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
