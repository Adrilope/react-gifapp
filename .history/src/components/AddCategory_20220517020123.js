import React from 'react'

export const AddCategory = () => {

    const 

    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
