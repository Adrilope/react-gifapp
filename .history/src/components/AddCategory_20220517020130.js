import React from 'react'

export const AddCategory = () => {

    useS
    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
