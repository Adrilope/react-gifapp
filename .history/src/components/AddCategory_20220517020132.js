import React from 'react'

export const AddCategory = () => {

    const [first, setfirst] = useState(second)
    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
