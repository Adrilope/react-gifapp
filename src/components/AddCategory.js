import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()  

        if (inputValue.trim().length > 2) {
            onNewCategory(inputValue.trim())
            setinputValue('')
        }
    }


    return (
        <form onSubmit={ handleSubmit }>
            <p>{inputValue}</p>
            <input 
                placeholder='Search gifs'
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired
}
