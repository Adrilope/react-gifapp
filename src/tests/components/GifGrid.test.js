import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
// indicamos donde estan los datos que vamos a mockear
jest.mock('../../hooks/useFetchGifs')


describe('Tests with <GifGrid />', () => { 
    const category = 'Bad Bunny'


    test('should render GifGrid correctly', () => { 
        // mockeamos los datos necesarios para simular la situcion de este test
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot()
    })


    test('should show the items when they`re loaded', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category}/>)

        // expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
        
    })

})