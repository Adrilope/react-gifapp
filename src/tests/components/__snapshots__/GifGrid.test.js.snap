// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests with <GifGrid /> should render GifGrid correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Bad Bunny
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
